src/Service/Resolvers/Sql Query Structures/SqlUpsertQueryStructure.cs

```diff
@@ -88,6 +88,17 @@ public class SqlUpsertQueryStructure : BaseSqlQueryStructure
             // Populates the UpsertQueryStructure with UPDATE and INSERT column:value metadata
             PopulateColumns(mutationParams, sourceDefinition, isIncrementalUpdate: incrementalUpdate);
 
+            if (FieldsReferencedInDbPolicyForCreateAction.Count > 0)
+            {
+                // If the size of this set FieldsReferencedInDbPolicyForCreateAction is 0,
+                // it implies that all the fields referenced in the database policy for create action are being included in the insert statement, and we are good.
+                // However, if the size is non-zero, we throw a bad request exception.
+                throw new DataApiBuilderException(
+                    message: "One or more fields referenced by the database policy are not present in the request.",
+                    statusCode: HttpStatusCode.BadRequest,
```

Comment:
>statusCode: HttpStatusCode.BadRequest, [](http://example.com/codeflow?start=20&length=38)

Because db policies are an extension of authorization permissions, should this be "Forbidden" because of a failure to meet the requirement of authZ? 

Bad Request would be more applicable If the user provides invalid column names